tools/nl/embeddings/build_embeddings_v2.py

```diff
@@ -0,0 +1,286 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Build the embeddings index by concatenating various inputs."""
+
+# TODO: This file should be renamed build_embeddings.py once it is ready to
+# replace the existing build_embeddings.py file.
+
+from dataclasses import dataclass
+import datetime as datetime
+import os
+from typing import Dict, List, Tuple
+
+from absl import app
+from absl import flags
+from google.cloud import storage
+import gspread
+import pandas as pd
+from sentence_transformers import SentenceTransformer
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('model_name', 'all-MiniLM-L6-v2', 'Model name')
+flags.DEFINE_string('bucket_name', 'datcom-nl-models', 'Storage bucket')
+
+flags.DEFINE_string('local_sheets_csv_filename', 'dc_nl_svs_curated',
+                    'Local Sheets CSV Filename')
+flags.DEFINE_string(
+    'sheets_url',
+    'https://docs.google.com/spreadsheets/d/1-QPDWqD131LcDTZ4y_nnqllh66W010HDdows1phyneU',
+    'Google Sheets Url for the latest SVs')
+flags.DEFINE_string('worksheet_name', 'Demo_SVs',
+                    'Worksheet name in the Google Sheets file')
+
+flags.DEFINE_string('palm_alternatives_filename', 'palm_alternatives',
+                    'PaLM alternative descriptions csv filename')
+flags.DEFINE_string('other_alternatives_filename', 'other_alternatives',
+                    'Other alternative descriptions csv filename')
+
+flags.DEFINE_string('merged_output_filename_prefix', 'embeddings',
+                    'Final merged/concatenated SVs and descriptions file.')
+
+# Col names in the input files/sheets.
+SHEETS_DCID_COL = 'Id'
+SHEETS_NAME_COL = 'Name'
+SHEETS_DESCRIPTION_COL = 'Description'
+SHEETS_ALTERNATIVES_COL = 'Curated_Alternatives'
+SHEETS_OVERRIDE_COL = 'Override_Alternatives'
+
+CSV_DCID_COL = 'Id'
+CSV_PALM_ALTERNATIVES_COL = 'PaLM_Generated_Alternatives'
+CSV_OTHER_ALTERNATIVES_COL = 'Other_Alternatives'
+
+# Col names in the concatenated dataframe.
+COL_ID = 'dcid'
+COL_ALTERNATIVES = 'sentence'
+
+# Setting to a very high number right for now.
+MAX_ALTERNATIVES_LIMIT = 50
```

Comment:
Curious why so high?

Reply:
just so that for now, we maintain parity with the existing embeddings and don't enforce a limit.